sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -285,7 +285,7 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
 
     // When we are regenerating the golden files, we don't need to set any config as they
     // all need to return the same result
-    if (regenerateGoldenFiles || !isTestWithConfigSets) {
+    if ((regenerateGoldenFiles && importedTestCaseName.isEmpty) || !isTestWithConfigSets) {
```

Comment:
ah seems we have 2 different kinds of settings now:
1. settings that shouldn't change result. We want to generate the golden files without the settings and run tests with settings.
2. setting that can change result.

Shall we have different directives for them? @maropu

Reply:
After a second thought, we only expect the `--SET` to change behavior if we import queries. So the change here LGTM.

@yaooqinn can you update the comment to explain it?

Reply:
BTW now `--SET` is upper case and `--import` is lower case. Seems better to always use upper case.

Reply:
also cc @wangyum , why do we need `isTestWithConfigSets` for thriftserver tests?

Reply:
OK, I will update the comments and change case for `--import`

Reply:
Now we are not able to use `--SET` and `--IMPORT` together for `ThriftServerQueryTestSuite`, for the `--SET` will always be ignored. Is there a chance for `--SET` + `--IMPORT` working for `ThriftServerQueryTestSuite`？

Reply:
@cloud-fan To reduce the test time. The `mvn` test often timed out at that time: https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-2.7/

Reply:
do you mean setting config is expensive?

Reply:
It will run `configSets` times:
https://github.com/apache/spark/blob/ec5d698d99634e5bb8fc7b0fa1c270dd67c129c8/sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala#L287-L298

Reply:
ah i see. But if there is only one config, it should be OK?

Reply:
ah that's a problem. We should think about how to reduce test time.

Reply:
Oh, I'm late.  The change looks reasonable...